Rick-and-Morty/Rick And Morty/Repositories/CharacterRepository.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+protocol CharacterRepositoryProtocol {
+    func getCharacters(completion: @escaping (([Character]) -> Void))
+}
+
+final class CharacterRepository: CharacterRepositoryProtocol {
+    private var characterPageURL: URL? = RickAndMortyService.baseURL.appendingPathComponent("character")
+    
+    private let rickAndMortyService: RickAndMortyServiceProtocol = RickAndMortyService()
+    
+    func getCharacters(completion: @escaping (([Character]) -> Void)) {
+        
+        if let url = characterPageURL {
+            rickAndMortyService.fetchData(url: url) { (characterList: CharactersList) in
+                if let nextURLString = characterList.info.next {
+                    self.characterPageURL = URL(string: nextURLString)!
```

Comment:
maybe add another `let` to this `if let` to remove the ! in `URL(string: nextURLString)!`